crates/nu-command/src/filters/where_.rs

```diff
@@ -10,13 +10,21 @@ impl Command for Where {
     }
 
     fn description(&self) -> &str {
-        "Filter values based on a row condition."
+        "Filter values of an input list based on a condition."
     }
 
     fn extra_description(&self) -> &str {
-        r#"This command works similar to 'filter' but allows extra shorthands for working with
-tables, known as "row conditions". On the other hand, reading the condition from a variable is
-not supported."#
+        r#"A condition is evaluated for each element of the input, and only elements which meet the condition are included in the output.
+
+A condition can be either a "row condition" or a closure. A row condition is a special short-hand syntax to makes accessing fields easier.
+Each element of the input can be accessed through the $it variable.
+
+On the left hand side of a row condition, any field name is automatically expanded to use $it.
+For example, `where type == dir` is equivalent to `where $it.type == dir`. This expansion does not happen when passing a subexpression or closure to where.
+
+When using a closure, the element is passed as an argument and as pipeline input to the closure.
```

Comment:
Maybe worth stating that the `$it` variable does not exist in the closure context?

Reply:
good idea